### PR TITLE
Fix progress calculations

### LINKED ISSUE #850: Progress bar renders as 100% at 0% progress for a planting
http://staging.growstuff.org/plantings/clockwerx-example-corn-salad
 * Plant from seed
 * 11 lettuces

Expected:
Progress bar renders 0% progress

Actual:
Progress bar renders 100% progress, which is styled progress-bar progress-bar-danger - it looks like a random red bar.

Suggestion:
 * Better tie together the bar,  progress %, and days until maturity
 * Skip rendering progress bars if we don't know the days until maturity

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Fixes #850 #849 

 * Prevents negative %
 * Prevents showing 100% growth, progress bar when growth isn't known
 * Adds test coverage and fixes previously wrong test coverage

### CODE REVIEW

File: app/views/plantings/_planting_progress.html.haml [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-	= render partial: "plantings/progress_bar", locals: {status: "danger", progress: "100%"}
+	= "Progress: Not calculated, days before maturity unknown"
 - else
-	- if (percent = (((DateTime.now - planting.planted_at)/planting.days_before_maturity*100).to_i)) >= 100
-		- percent = 100
-	= "Progress: #{percent}%"
-	= render partial: "plantings/progress_bar", locals: {status: "success", progress: "#{percent}%"}
\ No newline at end of file
+	= "Progress: #{planting.percentage_grown}%"
+	= render partial: "plantings/progress_bar", locals: {status: "success", progress: "#{planting.percentage_grown}%"}
```
Comment (cesy) [inline app/views/plantings/_planting_progress.html.haml]: Can we fix the missing newline while we're at it, please?

File: app/views/plantings/_thumbnail.html.haml [hunk 2]
```diff
@@ -38,4 +38,4 @@
       %dd= "#{display_days_before_maturity(planting)}"
 
   .col-xs-9.col-md-8
-    = render partial: 'plantings/planting_progress', locals: {:planting => planting}
\ No newline at end of file
+    = render partial: 'plantings/planting_progress', locals: {planting: planting}
```
Comment (cesy) [inline app/views/plantings/_thumbnail.html.haml]: And another newline here. Hooray for fixing to new syntax while going through! :)

File: spec/models/planting_spec.rb [hunk 3]
```diff
@@ -37,6 +37,48 @@
...
+      @planting.percentage_grown(Date.today).should be 40
+    end
+
+    it 'should not be calculated for unplanted plantings' do
+      @planting = FactoryGirl.build(:planting, planted_at: nil)
+
+      @planting.planted?.should be false
+      @planting.percentage_grown.should be nil
+    end
+    it 'should not be calculated for plantings with an unknown days before maturity' do
```
Comment (cesy) [inline spec/models/planting_spec.rb]: line break missing before this

**Comment by cesy** [comment]
Looks good, just a few minor line comments.

**Comment by CloCkWeRX** [comment]
Those should be fixed

**Comment by cesy** [comment]
Thanks!

**Pull request merged**